great_tables/_gt_data.py

```diff
@@ -396,6 +396,47 @@ def reorder(self, vars: list[str]) -> Self:
 
         return self[new_order]
 
+    # Get a mutated version of the Boxhead that:
+    # - removes hidden columns
+    # - places the stub column in the correct position
+    # - places the row group column in the correct position (if option is set for that)
+    def final_columns(self, options: Options) -> Self:
```

Comment:
What if rather than...

* get visible columns
* get stub
* if stub exists, remove from visible columns and create a new list of columns with stub at the front
* get group column
* if group column exists do something similar (shifting around columns)

You constructed the final column list directly, so something like:

* get stub column if it exists and append to list
* get row group column if it exists (and option dictates its a column) and append to list
* extend list with remaining columns (those marked ColInfoTypeEnum.default?)

This way the code reads in the same order as the final columns returned